### PR TITLE
Add "help" CLI option

### PR DESCRIPTION
**Pull request by UnBeatWaterGH**
Nice to have as not everyone wants to look at the code to figure out the CLI options.

I hope I documented -ab: properly.

### CODE REVIEW

**Comment by matt713** [comment]
IMHO, good idea, but I just want to suggest to change: "Use CM300 ROM set." to "Use CM300/SCC1 ROM set.", because maybe it is already not clear for some users that CM300 and SCC1 (not SCC1A - it's different!) use the same ROM set.

**Comment by pachuco** [comment]
Yis, nice.
> I hope I documented -ab: properly

The way it works is that you can feed it first parameter or both.
-ab:4096
-ab:4096:2

Also, I'd change page_number to num_pages or similar to suggest quantity of pages instead of an index.

**Comment by UnBeatWaterGH** [comment]
> Also, I'd change page_number to num_pages or similar to suggest quantity of pages instead of an index.

Do you think "page_amount" would work? I'd prefer not using too much shorthand in parameters.

File: src/mcu.cpp [hunk 1]
```diff
@@ -1351,6 +1351,26 @@ int main(int argc, char *argv[])
...
+            {
+                printf("Usage: %s [options]\n", argv[0]);
+                printf("Options:\n");
+                printf("  -h, -help, --help              Display this information.\n");
+                printf("\n");
+                printf("  -p:<port_number>               Set MIDI port.\n", argv[0]);
+                printf("  -a:<device_number>             Set Audio Device index.\n", argv[0]);
+                printf("  -ab:<page_size>:[page_amount]  Set Audio Buffer size.\n", argv[0]);
+                printf("\n");
+                printf("  -mk2                           Use MK2 ROM set.\n");
```
Comment (nukeykt) [inline src/mcu.cpp:1364]: `SC-55mk2`, `SC-55mk1`, `SC-55st`, `CM-300/SCC-1` and `JV-880` would be preffered names

File: src/mcu.cpp [hunk 2]
```diff
@@ -1351,6 +1351,26 @@ int main(int argc, char *argv[])
...
             }
+            else if (!strcmp(argv[i], "-h") || !strcmp(argv[i], "-help") || !strcmp(argv[i], "--help"))
+            {
+                printf("Usage: %s [options]\n", argv[0]);
+                printf("Options:\n");
+                printf("  -h, -help, --help              Display this information.\n");
+                printf("\n");
+                printf("  -p:<port_number>               Set MIDI port.\n", argv[0]);
+                printf("  -a:<device_number>             Set Audio Device index.\n", argv[0]);
+                printf("  -ab:<page_size>:[page_amount]  Set Audio Buffer size.\n", argv[0]);
```
Comment (nukeykt) [inline src/mcu.cpp:1362]: page_count might be better term

File: src/mcu.cpp [hunk 3]
```diff
@@ -1351,6 +1351,26 @@ int main(int argc, char *argv[])
             {
                 resetType = ResetType::GM_RESET;
             }
+            else if (!strcmp(argv[i], "-h") || !strcmp(argv[i], "-help") || !strcmp(argv[i], "--help"))
+            {
+                printf("Usage: %s [options]\n", argv[0]);
+                printf("Options:\n");
+                printf("  -h, -help, --help              Display this information.\n");
+                printf("\n");
+                printf("  -p:<port_number>               Set MIDI port.\n", argv[0]);
```
Comment (nukeykt) [inline src/mcu.cpp:1360]: argv[0]?

File: src/mcu.cpp [hunk 4]
```diff
@@ -1351,6 +1351,26 @@ int main(int argc, char *argv[])
             {
                 resetType = ResetType::GM_RESET;
             }
+            else if (!strcmp(argv[i], "-h") || !strcmp(argv[i], "-help") || !strcmp(argv[i], "--help"))
+            {
+                printf("Usage: %s [options]\n", argv[0]);
```
Comment (nukeykt) [inline src/mcu.cpp:1356]: not sure if argv[0] is right here, it may have full path. simply `nuked-sc55` might be better

Comment (UnBeatWaterGH) [reply src/mcu.cpp]: we probably want to do something similar to what GCC does, where we print out the executable's name instead of hard coding.

**Pull request merged**